src/prefect_server/utilities/collections.py

```diff
@@ -19,3 +20,61 @@ def chunked_iterable(iterable: Iterable, size: int):
         if not chunk:
             break
         yield chunk
+
+
+class TimedUniqueValueStore:
+    """
+    Stores unique values for a specific amount of time.
+
+    Args:
+        - duration (float, int): the number of seconds to store a value before removing it
```

Comment:
Usually we try to include the unit so it's clear when you call it e.g. `duration_seconds`